helpers/auth_x11.c

```diff
@@ -77,8 +81,86 @@ int prompt_timeout;
 //! Extra line spacing.
 #define LINE_SPACING 4
 
-//! Whether password display should hide the length.
-int paranoid_password;
+//! Deprecated flag for setting Whether password display should hide the length.
+int paranoid_password_flag;
+
+//! Updated flag for password display choice
+const char * password_prompt_flag;
+
+//! Actual password prompt selected
+enum PasswordPrompt {
+  PASSWORD_PROMPT_CURSOR,
+  PASSWORD_PROMPT_ASTERISKS,
+  PASSWORD_PROMPT_HIDDEN,
+  PASSWORD_PROMPT_DISCO,
+  PASSWORD_PROMPT_EMOJI,
+  PASSWORD_PROMPT_EMOTICON,
+  PASSWORD_PROMPT_KAOMOJI,
+
+  PASSWORD_PROMPT_MAX_VALUE = PASSWORD_PROMPT_KAOMOJI,
+};
+const char *PasswordPromptStrings[] = {
+  /* PASSWORD_PROMPT_CURSOR= */ "cursor",
+  /* PASSWORD_PROMPT_ASTERISKS= */ "asterisks",
+  /* PASSWORD_PROMPT_HIDDEN= */ "hidden",
+  /* PASSWORD_PROMPT_DISCO= */ "disco",
+  /* PASSWORD_PROMPT_EMOJI= */ "emoji",
+  /* PASSWORD_PROMPT_EMOTICON= */ "emoticon",
+  /* PASSWORD_PROMPT_KAOMOJI= */ "kaomoji",
+};
+
+enum PasswordPrompt password_prompt;
+
+// A disco password is composed of multiple disco_dancers (each selected at random from the array), joined by the disco_combiner
+const char * disco_combiner = " ♪ ";
+// Note: the disco_dancers MUST all have the same length
+const char * disco_dancers[] = {
+  "┏(･o･)┛",
+  "┗(･o･)┓",
+};
+
+// Emojis to display in emoji mode. The length of the array must be equal to PARANOID_PASSWORD_LENGTH.
+// List taken from the top items in http://emojitracker.com/
+// The first item is always display in an empty prompt (before typing in the password)
+const char * emojis[] = {
+  "_____", "\U0001f602", "\U00002764", "\U0000267b",
+  "\U0001f60d", "\U00002665", "\U0001f62d", "\U0001f60a",
+  "\U0001f612", "\U0001f495", "\U0001f618", "\U0001f629",
+  "\U0000263a", "\U0001f44c", "\U0001f614", "\U0001f601",
+  "\U0001f60f", "\U0001f609", "\U0001f44d", "\U00002b05",
+  "\U0001f605", "\U0001f64f", "\U0001f60c", "\U0001f622",
+  "\U0001f440", "\U0001f494", "\U0001f60e", "\U0001f3b6",
+  "\U0001f499", "\U0001f49c", "\U0001f64c", "\U0001f633",
+};
+_Static_assert(sizeof(emojis) / sizeof(*emojis) == PARANOID_PASSWORD_LENGTH, "Emojis array size must be equal to PARANOID_PASSWORD_LENGTH");
```

Comment:
Given this is our only use of C11 anywhere, please put the _Static_asserts into an

```
#if __STDC_VERSION__ >= 201112L
```

block. Thanks!

Reply:
How about a macro?

```
#if __STDC_VERSION__ >= 201112L
#define ASSERT(state, message) _Static_assert(state, message)
#else
#define ASSERT(state, message)
#endif
```

Implemented in c1cfe35 for your review.

Reply:
Yeah, that will do.